internal/querynode/meta_replica.go

```diff
@@ -853,6 +855,20 @@ func (replica *metaReplica) getSealedSegments() []*Segment {
 	return ret
 }
 
+// removeCollectionVDeltaChannel remove vdelta channel replica info from collection.
+func (replica *metaReplica) removeCollectionVDeltaChannel(collectionID UniqueID, vDeltaChannel string) {
+	replica.mu.RLock()
+	defer replica.mu.RUnlock()
+
+	coll, ok := replica.collections[collectionID]
+	if !ok {
+		// if collection already release, that's fine and just return
+		return
+	}
+
+	coll.removeVDeltaChannel(vDeltaChannel)
```

Comment:
lock collection here?

Reply:
collection has `channelMut` for channel modification, which is used in the `removeVDeltaChannel`